python/lib/core/dmod/core/common/helper_functions.py

```diff
@@ -27,31 +32,31 @@ def get_mro_names(value) -> list[str]:
 
 
 def is_integer(value) -> bool:
-    if type(value) != type:
-        value = value.__class__
-
-    mro = get_mro_names(value)
+    if numpy and isinstance(value, numpy.integer):
+        return True
 
-    module_name = value.__module__ if hasattr(value, "__module__") else ''
+    loaded_integer_types = [int] + get_subclasses(int)
 
-    if module_name.startswith("numpy"):
-        return "integer" in mro
+    if numpy:
+        loaded_integer_types.extend(
+            [numpy.integer] + get_subclasses(numpy.integer)
+        )
 
-    return isinstance(value, int)
+    return value in loaded_integer_types if type(value) == type else isinstance(value, int)
```

Comment:
I think this function can be reduced to:

`return isinstance(value, int) if numpy is None else isinstance(value, (int, np.integer))`

Reply:
Only reason not to do that is to cover the `is_integer(int)` case you presented in the other ticket.